### ISSUE TITLE
Add more POI types to searches for vegetarian/vegan options

### ISSUE THREAD

**Issue by une-abeille-osm** [organicmaps/organicmaps#8612]
I love the fact that I can search for vegan restaurants with OrganicMaps, but I'm sad that some businesses do not show up in the search.

Additional tags that could benefit from being included in this search:
`shop=bakery`
`shop=pastry`
`amenity=ice_cream`

I'm sure there are others I haven't encountered. Basically anything to do with food!

**Comment by RedAuburn** [comment]
can reproduce:

[Vida Bakery](https://www.openstreetmap.org/way/271645795) at the bottom should be matched
<img width="300px" src="https://github.com/organicmaps/organicmaps/assets/26939824/9067d3bb-06c8-40d1-a08c-99db01dc45ec"/>

**Comment by RedAuburn** [comment]
https://github.com/organicmaps/organicmaps/blob/52b616e18cf2f5c351f9f4f59493fb52adcb43ed/search/processor.cpp#L278-L283

IsEatChecker currently goes off a hardcoded list:

https://github.com/organicmaps/organicmaps/blob/52b616e18cf2f5c351f9f4f59493fb52adcb43ed/indexer/ftypes_matcher.cpp#L657-L668

either we could make ftypes_matcher use the types tagged with @category_eat in data/categories.txt, or just drop the type checker for cuisines? a clothes shop isn't gonna be tagged with cusines etc, so i think it'd be fine.

**Comment by une-abeille-osm** [comment]
> a clothes shop isn't gonna be tagged with cusines etc, so i think it'd be fine.

Well, actually, I've tagged a few things like tattoo shops with the `diet:vegan` tag (because I couldn't find anything more appropriate), so you might want to make sure that things like that don't show up in the search results. :laughing:

**Comment by RedAuburn** [comment]
looking through categories.txt, only ice_cream, pastry and bakery seem to be missing, so i think just adding them to the checker is easiest :)

**Comment by biodranik** [comment]
I'm not convinced with bakery or ice cream displayed in the search results as a place where you can eat. If someone wants a bakery, he/she will search for a bakery/pastry.

If the issue is about searching for _anything_ with a vegan/vegetarian cuisine, then it should be fixed appropriately, not by mixing _real food_ places with sweets and pastry.

**Comment by RedAuburn** [comment]
mm i guess so, in that case https://github.com/organicmaps/organicmaps/issues/8620 is the answer

**Comment by une-abeille-osm** [comment]
> I'm not convinced that a bakery or ice cream should be displayed in the search results as a place where you can eat. If someone wants a bakery, he/she will search for a bakery/pastry.
> 
> If the issue is about searching for _anything_ with vegan/vegetarian cuisine, then it should be fixed appropriately, not by mixing _real food_ places with sweets and pastries.
> 
> A better search can also narrow results if "vegan bakery" or "vegan food" is typed.

I don't see much of a difference between what's offered at a bakery and "real food" at a place like a café. Bakeries typically have options that would be fine for lunch on the go, and in many cases differ from cafés only insofar that the food offerings are baked on-premises.

If I am searching *just* for "vegan," I generally want to see what is closest out of all available options. If I wanted only a substantial meal, I would probably search for "vegan restaurant."

**Comment by une-abeille-osm** [comment]
> mm i guess so, in that case #8620 is the answer

This would be a big upgrade!

**Comment by eisams** [comment]
> I don't see much of a difference between what's offered at a bakery and "real food" at a place like a café. Bakeries typically have options that would be fine for lunch on the go, and in many cases differ from cafés only insofar that the food offerings are baked on-premises.

Agree, I think most bakeries in Norway are hybrid cafés (may be different elsewhere in the world)
